### PR TITLE
feat: disable dp

### PR DESCRIPTION
**Pull request by Marc-Roig**
### What does it do?

Describe the technical changes you did.

### Why is it needed?

Describe the issue you are solving.

### How to test it?

Provide information about the environment and the path to verify the behaviour.

### Related issue(s)/PR(s)

Let us know if this is related to any issue/pull request

### CODE REVIEW

File: api-tests/core/strapi/document-service/dp/basic-no-dp.test.api.ts [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+const testInTransaction = (...args: Parameters<jest.It>) => {
+  if (args.length > 1) {
+    return it(
+      args[0], // name
+      wrapInTransaction(args[1]), // fn
+      args[2] // timeout
+    );
+  }
+  return it(...args);
+};
```
Comment (Marc-Roig) [inline api-tests/core/strapi/document-service/dp/basic-no-dp.test.api.ts:14]: will actually remove this from here and do another separate pr for it

File: api-tests/core/strapi/document-service/relations/no-dp.test.api.ts [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+          populate: xToOneRelations,
+        });
+
+        // Products should be connected to both draft and published version
+        xToOneRelations.forEach((relation) => {
+          expect(shop[relation]).toMatchObject([
+            { name: 'Skate-En', locale: 'en', publishedAt: null },
+            { name: 'Skate-En', locale: 'en', publishedAt: expect.any(String) },
+          ]);
+        });
```
Comment (Marc-Roig) [inline api-tests/core/strapi/document-service/relations/no-dp.test.api.ts:171]: @alexandrebodin I would like to review the set of tests for relations together 

Specially, I would like to discuss the approach on return values of "oneToOne" relations, if a one to one relation can return a published and draft version, it has to be an array. 

That will be a breaking change, and I wonder if it just would make sense (in xToOne relations) to always return a single relation at a time (draft or publish and pick a default)

File: packages/core/content-type-builder/server/src/controllers/validation/content-type.ts [hunk 3]
```diff
@@ -22,6 +22,7 @@ export type CreateContentTypeInput = {
   displayName: Schema.ContentTypeInfo['displayName'];
   description: Schema.ContentTypeInfo['description'];
   options?: Schema.Options;
+  withDraftAndPublish?: Schema.Options['withDraftAndPublish'];
```
Comment (alexandrebodin) [inline packages/core/content-type-builder/server/src/controllers/validation/content-type.ts:25]: use the v4 naming to avoid a not so necessary BC

File: packages/core/content-type-builder/server/src/services/schema-builder/content-type-builder.ts [hunk 4]
```diff
@@ -101,7 +101,10 @@ export default function createComponentBuilder() {
           displayName: infos.displayName,
           description: infos.description,
         })
-        .set('options', infos.options ?? {})
+        .set('options', {
+          ...(infos.options ?? {}),
+          withDraftAndPublish: infos.withDraftAndPublish ?? true,
```
Comment (alexandrebodin) [inline packages/core/content-type-builder/server/src/services/schema-builder/content-type-builder.ts:106]: No necessary if we default to true in on loading (in the domain/content-types)

**Review by alexandrebodin — COMMENTED** [review]
small comments just to remember :)

**Comment by Marc-Roig** [comment]
Will create other PRs for relations, I have checked and frontend does receive the default value of draft and publish if it is not set in the schema 👍 cc @alexandrebodin

**Comment by joshuaellis** [comment]
Do you think it's better to build the whole feature on this branch like we are for relations? or merge incrementally, honestly i'd opt for the former considering they're small features 👀

**Review by joshuaellis — APPROVED** [review]
Approving because all the PRs were approved by the appropriate individuals
